### DISCUSSION TITLE
Remove time from date column

### DISCUSSION THREAD

**Discussion by GridTrekkor** [exceljs/exceljs#1940] [Q&A]
I'm using a `numFmt` like this to format a date in Excel as an actual Date: `column.style = { numFmt: "mm/dd/yyyy" };`

The date displays correctly in the sheet, but when clicking the value, I also see a timestamp.

In the sheet:

![image](https://user-images.githubusercontent.com/5239633/147140061-672215b8-463e-441a-be6f-02efed54e034.png)

When clicking the value:

![image](https://user-images.githubusercontent.com/5239633/147140097-11705dae-1c62-4620-b7d8-fa027b74fbef.png)

So `04/22/1938` in the sheet has an actual value of `04/22/1938  06:00:00`

How can I remove the time component so the value is actually `04/22/1938`?

**Comment by ghost** [comment]
Hey @GridTrekkor!

Did you find a solution?

**Reply by GridTrekkor** [reply]
Hi @harryn - haven't yet found a solution.  I suspect it happens because all JS dates have a time component, and it may or may not be removable by exceljs.

**Comment by skypesky** [comment]
@GridTrekkor @harryn 
How to reproduce this error with code?

**Comment by ghost** [comment]
Seems it happens when your date has time part greater than 0. So, any date that is not at midnight, will show time part in the value box (cell is still displayed correctly based on format).

In addition, if your environment is not UTC, then this will cause time to appear also, even though your date could be at midnight. I expect, that @GridTrekkor's timezone is UTC-6, because additional 6 hours seems to be added to the date.

For now, I am dealing with it this way:
```js
const correctDate = moment.utc('1938-04-22') // use UTC timezone instead of your local/default
.startOf('day') // kind of "removes" time as it will be 00:00:00 (midnight)
.toDate(); // convert to Date object
```

If you want to reproduce it yourself:
```js
import ExcelJS from "exceljs";
import moment from "moment";

const workbook = new ExcelJS.Workbook();
const sheet = workbook.addWorksheet('Debug');
sheet.columns = [{
  header: 'debug',
  key: 'debug',
  style: { numFmt: 'mm/dd/yy' }
}]

sheet.addRows([
  {debug: new Date()},// should show your current time in the value box
  {debug: new Date('1938-04-22')},// UTC-6 should show 06:00:00 time part in the value box (and for example UTC+2 would show previous date and 22:00:00)
  {debug: moment.utc('1938-04-22').startOf('day').toDate()},// no time part when your timezone is UTC-X
  {debug: moment.utc('1938-04-22').add(1, 'day').startOf('day').toDate()},// no time part when your timezone is UTC+X
]);

await workbook.xlsx.writeFile(`testing.xlsx`);
```

**Reply by OmniSliver** [reply]
Thanks! Using a Date at UTC midnight worked for me. Using moment is not necessary though, I used `new Date('1938-04-22')`
